Classes/class-serviceworker.php

```diff
@@ -44,34 +54,15 @@ public function ssl_error_notice() {
 	}
 
 	public function add_to_header() {
-
-		$url = untrailingslashit( get_home_url() ) . $this->sw_url;
 		?>
 		<script type="text/javascript" id="serviceworker">
-			if ('serviceWorker' in navigator) {
-				if (location.protocol !== 'https:') {
-					console.log('[Progressive WordPress] <?php echo addslashes( __( 'Your site needs to be served via HTTPS to use Progressive Web App functionality', 'pwp' ) ); ?>');
-				} else {
-					window.addEventListener('load', function () {
-						navigator.serviceWorker.register('<?php echo pwp_register_url( $url ); ?>')
-							.then(function (registration) {
-								registration.update();
-							})
-							.catch(function (error) {
-								console.log('[Progressive WordPress] <?php echo addslashes( __( 'Registration failed', 'pwp' ) ); ?>: ' + error);
-							});
-					});
-				}
-			} else {
-				console.log('[Progressive WordPress] <?php echo addslashes( __( 'Your browser does not support Progressive Web App functionality', 'pwp' ) ); ?>');
-			}
 			<?php
 			if ( pwp_get_setting( 'pwp-force-deregister-sw' ) ) {
```

Comment:
Is this going to be relevant in the new core SW API? When the service worker is de-enqueued then the core SW should prompt to re-install automatically, right?

Reply:
Hmm. In the context of this plugin I guess it's safe to remove these lines anyway since it's unregistering any service worker that does not match with the original URL but we're not using the original URL anymore. 

Generally thinking, when the SW gets unregistered then the SW API registers it again, ¿is that what you were thinking of?, however, if the unregistering is happening on every page load after the registering then I guess the SW wouldn't get registered again. Not sure if that's something we could avoid.

Reply:
I'm not sure. I guess I don't know what the purpose is for unregistering the service worker in the plugin.

@nico-martin Can you share insights into why the service worker can be forcibly-deregistered? In the context of a core SW API, would this become irrelevant since presumably the plugin could just cease to register the `progressive-wp-sw` script?

Reply:
If you have multiple ServiceWorkers registered which handle fetch events, there might be a conflict between those events.
I ran into this issue quite often when people tried multiple PWA-Plugins. It was quite hard for me to find the error. Because when I visited the site, I had only my serviceworker registered. But since they tried multiple plugins, and even if those plugin were not longer active, the ServiceWorkers were still registered in their browser.

Unregistering all other ServiceWorkers allowes me to get rid of old ServiceWorkers which are still reistered because a browser visited the site when an other plugin was in use.

Reply:
Yes. Thats right. But we should somehow make sure the plugins and themes that currently have their own SW add an unregister script as soon as they switch to the ServiceWorker API.

Reply:
Yes, that is be something they would be responsible for.